js/ajax.php

```diff
@@ -30,94 +30,102 @@
 $return['error']   = false;
 $return['message'] = '';
 
-switch( $_POST['sailthru_action'] )
-{
-
-	case "add_subscriber":
-		$email = trim( $_POST['email'] );
-		if ( ! filter_var( $email, FILTER_VALIDATE_EMAIL ) || empty( $email ) ) {
-			$return['error']   = true;
-			$return['message'] = 'Please enter a valid email address.';
-		} else {
-			$email = filter_var( $email, FILTER_VALIDATE_EMAIL );
-		}
-
-		if ( isset( $_POST['first_name'] ) && !empty( $_POST['first_name'] ) ){
-			$first_name = filter_var(trim($_POST['first_name']), FILTER_SANITIZE_STRING );
-		} else {
-			$first_name = '';
-		}
-
-		if ( isset( $_POST['last_name'] ) && !empty( $_POST['last_name'] ) ){
-			$last_name = filter_var( trim( $_POST['last_name'] ), FILTER_SANITIZE_STRING ) ;
-		} else {
-			$last_name = '';
-		}
-
-		if ( $first_name || $last_name ) {
-
-			$options = array(
-				'vars' => array(
-					'first_name'	=> $first_name,
-					'last_name'		=> $last_name,
-				)
-			);
+if( isset( $_POST['sailthru_action'] ) ) {
+
+	switch( $_POST['sailthru_action'] )
+	{
+
+		case "add_subscriber":
+			if ( isset( $_POST['email'] ) ){
+				$email = trim( sanitize_email( $_POST['email'] ) );
+				if ( ! filter_var( $email, FILTER_VALIDATE_EMAIL ) || empty( $email ) ) {
+					$return['error']   = true;
+					$return['message'] = 'Please enter a valid email address.';
+				} else {
```

Comment:
**suggestion:** If `isset( $_POST['email'] )` is false, I think you want to same behavior as when `empty( $email )`. What you have right now is bypassing the error message.